### DISCUSSION TITLE
Discontinuing the use of issues for comments, questions and announcements

### DISCUSSION THREAD

**Discussion by JonathanGregory** [cf-convention/discuss#335] [Comments and ideas for changing CF]
Dear all

At the CF committee meeting last month, we talked about the [CF GitHub Discussions](https://github.com/orgs/cf-convention/discussions) forum instituted a few months ago. Since it is working fine, in our opinion, we decided to _propose that_ **all comments, questions and announcements in future must be started as a new Discussion in the forum** (it has those three categories at present).

Discussions can (and generally should) be labelled, to indicate what they're about. Thus, you can indicate a question or comment is about a standard name or conventions by attaching the appropriate label. If you have a definite proposal for a standard name or change to the convention, you can open an issue in the appropriate repo to make the proposal. If you're not sure whether or what to propose, it would be appropriate to start with a Discussion about it.

Does anyone have concerns or comments about this proposed change?

If it is agreed, we will **discontinue** use of new issues for announcements and questions in the `discuss` repository, and for questions in the `cf-conventions` repository. We will keep the old issues, of course. With this change and the one [announced by Alison](https://github.com/orgs/cf-convention/discussions/334), the `discuss` repository will no longer be used for new issues.

Best wishes

Jonathan

**Comment by DocOtak** [comment]
Fantastic! Full support from me for this.

Should we try switching to the Q&A format for the... Q&A category? I agree that the voting mechanic might not be desirable, but also having a clearly marked answer to a question might be worth it.

**Comment by JonathanGregory** [comment]
Dear Andrew @DocOtak

Thanks for your support, as well as for your advocating that we should introduce Discussions in the first place.

My opinion (as you probably already know) is that the Q&A mechanism isn't appropriate for CF questions. Usually there is only one answer, which is provided by one person, or arrived at by debate, which often involves clarifying the question. The answer can therefore be found by looking at the end of the Discussion, and doesn't have to be decided by voting. Like with other things in CF, it's a consensus. But that's just my opinion! A different consensus may emerge. :smiley:

I think we could make Discussion questions a more useful resource by (a) labelling them as `answered` when they have been, (b) linking or summarising particularly useful answered questions in the [FAQ](https://cfconventions.org/faq.html), which is a resource that needs revision but no-one has had time to attend to recently.

Best wishes

Jonathan

**Reply by DocOtak** [reply]
I think we should just give the actual Q&A format a trial period. Both the label and link solutions feel like reinventing this format anyway to me. It looks like Forum Moderators might have the ability to [mark a particular comment as the answer](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer). Which would a) show the question as answered, b) show a snippet of that answer right under the question but link to the answers location so full context is always there. The voting mechanism is obscure enough that I didn't realize it was a thing until reading the documentation. The "answered" status appears to be a built in status category rather than just a label. Check out the [xarray discussions Q&A section](https://github.com/pydata/xarray/discussions/categories/q-a) for an actively used example. I don't know if the original question asker can mark a reply as the answer themselves or if only a forum moderator can do that.

**Reply by ethanrd** [reply]
I agree with Andrew @DocOtak that we should look further at or even try out the Q&A format. Having the answer available immediately below the question seems like a big win for longer and complex discussions.

**Comment by JonathanGregory** [comment]
I could be convinced by examples of previous CF questions where this would help. Some CF questions are not really questions requiring an well-defined answer; they're really discussions, and we should change their category when we realise. I'm concerned because of my impression that some of them require a debate to find the answer. Upvoting might disrupt the flow and make it unintelligible.

**Reply by DocOtak** [reply]
It looks like even the format we are discussing this in right now supports upvoting. So I'm going to drop that as a concern I have about switching one of the categories to the Q&A format. As far as I can tell, the only thing we get by switch to Q&A as a format is the ability to mark a comment and it's chain of replies as an answer.

Some examples that are basically concluded:
* https://github.com/orgs/cf-convention/discussions/307 
  * Basically answered by themselves: https://github.com/orgs/cf-convention/discussions/307#discussioncomment-9127993
* https://github.com/orgs/cf-convention/discussions/324
  * Either https://github.com/orgs/cf-convention/discussions/324#discussioncomment-9672660
  * or https://github.com/orgs/cf-convention/discussions/324#discussioncomment-9705283
* https://github.com/orgs/cf-convention/discussions/318
  * Answered by the first reply: https://github.com/orgs/cf-convention/discussions/318#discussioncomment-9488788
* https://github.com/orgs/cf-convention/discussions/339
  * Answered by https://github.com/orgs/cf-convention/discussions/339#discussioncomment-10180137

**Comment by JonathanGregory** [comment]
Dear Andrew @DocOtak and @ethanrd

I take back my objection to upvoting, now I've understood that's not a permanent change to the discussion. My concern was that upvoting comments would reorder the discussion, and hence not make any sense. Actually it's just an alternative view to historical order, so that's fine. Sorry for my ignorance!

You're right that upvoting is possible in comments too, like this one. Thanks for pointing it out. I was misled by the GitHub instructions when I set up the Discussion categories:
![temp](https://github.com/user-attachments/assets/f4b2eda9-b482-4006-8cba-fab9f845f79a)
This seems to imply that upvoting only exists in Q&A. In fact, the unique functionality is "Mark as answered" and the Answered/Unanswered status. I agree that those could be useful.

Thanks for the examples. For three of them, the discussion is very short and the conclusion is therefore easy to see. In the other, which is long, the fullest answer is the last one. I suspect that the answer to a CF question is usually at the bottom, and might possibly be spread over several remarks near the bottom, which are individually useful ingredients. In such a case, a summary answer would be helpful. Summary answers are useful material for the FAQ, or the FAQ can be an index to them. Some curation is needed, in any case, which takes effort.

Best wishes

Jonathan

**Comment by JonathanGregory** [comment]
I've changed the Q&A category to the `Question / Answer` type of Discussion.

**Comment by JonathanGregory** [comment]
There haven't been any objections expressed in the last three weeks to discontinuing the use of issues for comments, questions and announcements, all of which would be started as GitHub Discussions after we've made this change. Therefore I will now open an issue to propose this change.

**Reply by DocOtak** [reply]
For maintainer purposes, is it documented which repository is the "host" repository for the organization level? This would be for things like templates and other github features we might want to utilize.

**Reply by JonathanGregory** [reply]
The `discuss` repo is the host for these organization-level discussions (I've mentioned that in the [issue](https://github.com/cf-convention/discuss/issues/346)). What sorts of templates can be used at the organization level?

**Reply by DocOtak** [reply]
It's just as diverse as for issue templates. https://docs.github.com/en/discussions/managing-discussions-for-your-community/creating-discussion-category-forms

**Reply by JonathanGregory** [reply]
I see. Yes, the category templates are in this repo as well, because it hosts the discussions.

**Comment by JonathanGregory** [comment]
No-one raised any concerns about the proposal in [`discuss` issue 346](https://github.com/cf-convention/discuss/issues/346) to discontinue the use of new issues in the `discuss` repo. Therefore that change is now agreed. I have closed the issue.

In accordance with this change, I have modified the issue configuration for `discuss` so that no new issues can be started there. If you try to open a new issue, you are directed to [Discussions](https://github.com/orgs/cf-convention/discussions) of the appropriate category.

I will prepare some proposed changes to the website to suit the new arrangement.

**Comment by JonathanGregory** [comment]
[Website issue 542](https://github.com/cf-convention/cf-convention.github.io/issues/542) proposes change to the website to suit the new arrangement, with a PR to implement them. ... Now merged, and issued closed.
